core/services/feeds/service.go

```diff
@@ -321,75 +335,39 @@ func (s *service) Start() error {
 		}
 
 		mgr := mgrs[0]
-
-		s.connect(mgr.URI, privkey, mgr.PublicKey, mgr.ID)
+		s.connectFeedManager(mgr, privkey)
 
 		return nil
 	})
 }
 
 func (s *service) Close() error {
 	return s.StopOnce("FeedsService", func() error {
-		// Close any blocking dials
-		s.connCtxCancel()
-		// Close any active connections
-		close(s.chDone)
+		// This blocks until it finishes
+		s.connMgr.Close()
 
-		s.wgDone.Wait()
 		return nil
 	})
 }
 
-// Connect attempts to establish a connection to the Feeds Manager.
-//
-// In the future we will connect to multiple Feeds Managers. Each `connect` call
-// will run in a separate goroutine. Closing the feeds service will shutdown
-// each goroutine.
-func (s *service) connect(uri string, privkey []byte, pubkey []byte, feedsManagerID int64) {
-	s.wgDone.Add(1)
-
-	go gracefulpanic.WrapRecover(func() {
-		defer s.wgDone.Done()
-
-		// Clean up context
-		defer s.connCtxCancel()
-
-		conn, err := wsrpc.DialWithContext(s.connCtx, uri,
-			wsrpc.WithTransportCreds(privkey, ed25519.PublicKey(pubkey)),
-			wsrpc.WithBlock(),
-		)
-		if err != nil {
-			// We only want to log if there was an error that did not occur
-			// from a context cancel.
-			if s.connCtx.Err() == nil {
-				logger.Infof("Error connecting to Feeds Manager server: %v", err)
-			}
-
-			return
-		}
-		defer conn.Close()
-
-		logger.Infow("[Feeds] Connected to Feeds Manager", "feedsManagerID", feedsManagerID)
-
-		// Initialize a new wsrpc client to make RPC calls
-		s.mu.Lock()
-		s.fmsClient = pb.NewFeedsManagerClient(conn)
-		s.mu.Unlock()
-
-		// Initialize RPC call handlers on the client connection
-		pb.RegisterNodeServiceServer(conn, &RPCHandlers{
-			feedsManagerID: feedsManagerID,
+// connectFeedManager connects to a feeds manager
+func (s *service) connectFeedManager(mgr FeedsManager, privkey []byte) {
```

Comment:
same thing here - I think we want the connection attempt to be scoped to the API request via its context

Reply:
The connection attempt runs in a go routine and will attempt to reconnect with a backoff, so that if the server is down it will reconnect when it comes back up.

Because of this we don't want to cancel this based on the request context.